### ISSUE TITLE
why this setting (minibatchsize=1) is slow?

### ISSUE THREAD

**Issue by Eomtaesan** [Microsoft/CNTK#897]
if minibatchsize is high, program is fast, but if it is low, speed is slow. why?

**Comment by Eomtaesan** [comment]
additional :
i dont want to use SGD
i want to set minibatchsize=1
why must use this?

**Comment by e-thereal** [comment]
I'm making a couple of assumption to be able to answer your question. First of, I'm assuming that you run your program on the GPU and I'm assuming that your input features are rather low-dimensional (e.g., small images), and that you are using some kind of neural network. In that case, processing small images does not utilize a modern GPU. In order to use all the cores and therefore to gain the maximum data throughput, you need to process multiple images at the same time. This is what is happening when you set the minibatchsize to a value greater than 1. In that case, multiple images are processed at the same time in parallel, which results in a better utilization of the available GPU cores and therefore a faster training speed.

You said you don't want to use SGD. What other optimization method do you want to use?

**Comment by Eomtaesan** [comment]
e-thereal
Thanks the answer,
My network is simple 1840:7360*6:1840, running on cpu_only
My # of training set is just 81.
I know this batch normalization is only work in batch time? or batch input?
I want to normalize my all input, so I want to set minibatchsize=1.
But I think that this setting occur
1. Some data are skipped by minibatchsize, but my training set is only 81, so I dont want to skip
2. In batch time, learning rate is modified to learning rate per sample * batchsize.

These reason make me think that is unuseful.

**Comment by e-thereal** [comment]
I haven't used CNTK for sequence learning or anything that involves time. Sorry I can't help you with that.

**Comment by cha-zhang** [comment]
You should not use batch normalization with minibatchsize=1. See https://github.com/Microsoft/CNTK/issues/905

**Comment by cha-zhang** [comment]
Close this issue since no response was given for a long time. Hope this issue has been resolved.
